test/unittest/db/ibis/test_end_2_end.py

```diff
@@ -0,0 +1,172 @@
+import ibis
+import pandas
+import PIL.Image
+import pytest
+from sklearn import svm
+import random
+import torchvision
+
+from superduperdb import superduper
+from superduperdb.db.ibis.schema import IbisSchema
+from superduperdb.ext.pillow.image import pil_image
+from superduperdb.ext.torch.model import TorchModel
+from superduperdb.ext.sklearn.model import Estimator, SklearnTrainingConfiguration
+
+
+chars = 'abcdefghijklmnopqrrstuvwxyz'
```

Comment:
You have two `r` s!!

You should instead use

https://docs.python.org/3/library/string.html#string.ascii_lowercase

:-)